library/core/src/iter/traits/iterator.rs

```diff
@@ -1825,10 +1825,18 @@ pub trait Iterator {
         Inspect::new(self, f)
     }
 
-    /// Borrows an iterator, rather than consuming it.
+    /// Creates a "by reference" adapter for this instance of `Iterator`.
     ///
-    /// This is useful to allow applying iterator adapters while still
-    /// retaining ownership of the original iterator.
+    /// Consuming method calls (direct or indirect calls to `next`)
+    /// on the "by reference" adapter will consume the original iterator,
+    /// but ownership-taking methods (those with a `self` parameter)
+    /// only take ownership of the "by reference" iterator.
+    ///
+    /// This is useful to allow applying ownership-taking methods,
```

Comment:
```suggestion
    /// This is useful to allow applying ownership-taking methods
```
:wink:

Reply:
I'd also give an example for such a method (perhaps `count`?), so people have a really clear understanding of what this could be used for.

Reply:
I've implemented your suggestion. Let me know if you think this is better.

Reply:
@rustbot ready